### ISSUE TITLE
Detect invalid arity situations with functions like `update` and `swap!`.

### ISSUE THREAD

**Issue by jakemcc** [clj-kondo/clj-kondo#2003]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

**Is your feature request related to a problem? Please describe.**

Sometimes the function passed to `update` has its signature changed (adding or removing an argument). This results in some invalid code unless the developer catches all the of uses of the changed function.

Example below:

```clojure
(defn two-args
  [coll x]
  (conj coll x))

(update {:a [1]} :a two-args) ;; uh-oh, wrong number of args to two-args but not flagged

(two-args []) ;; flagged as invalid-arity
```

**Describe the solution you'd like**

It seems like it would be possible to write a linter to detect this type of invalid arity situation. This could be applied to functions similar to `update` or `swap!`.

**Describe alternatives you've considered**

I haven't.

**Additional context**

[Thread in Slack asking if this seems possible](https://clojurians.slack.com/archives/CHY97NXE2/p1678465677346899)

I'm happy to take a look at this with some pointing in the right direction. Also happy to let someone familiar with the project do so.

**Comment by borkdude** [comment]
Can you make another issue for `swap!`?

<img width="977" alt="Screenshot 2023-03-11 at 10 44 46" src="https://user-images.githubusercontent.com/284934/224477245-db4611c6-f390-412b-8278-5e5eaebf6c10.png">

**Comment by borkdude** [comment]
Made the issue here: https://github.com/clj-kondo/clj-kondo/issues/2004

Perhaps you can check out the latest commits and implement `swap!` yourself if you like

**Comment by borkdude** [comment]
`swap!` is now also supported.
